napari/_vispy/experimental/texture_atlas.py

```diff
@@ -0,0 +1,203 @@
+"""TextureAtlas2D class.
+
+A texture atlas is a large texture that stores many smaller texture tiles.
+"""
+from collections import namedtuple
+from typing import Optional, Tuple
+
+import numpy as np
+from vispy.gloo import Texture2D
+
+# Two triangles which cover a [0..1, 0..1] quad.
+_QUAD = np.array(
+    [[0, 0], [1, 0], [1, 1], [0, 0], [1, 1], [0, 1]], dtype=np.float32,
+)
+
+# TexInfo is returned from TextureAtlas2D.add_tile() so the caller has the
+# texture coordinates to render each tile in the atlas.
+TexInfo = namedtuple('TexInfo', "tile_index tex_coord")
+
+
+class TextureAtlas2D(Texture2D):
```

Comment:
I think both `TextureAtlas2D` and `TiledImageVisual` are candidates. But our "tiled image visual" is much less than what David probably had in mind. But it seems useful as part of a bigger thing. A simple/solid building block rather than something too complicated .

Reply:
I added a design diagram above that I think explains how our `TiledImageVisual` works compared with maybe what David expected.